deepinv/physics/forward.py

```diff
@@ -63,13 +64,8 @@ def __mul__(self, other):  #  physics3 = physics1 \circ physics2
         :return: (:class:`deepinv.physics.Physics`) concatenated operator
 
         """
-        A = lambda x: self.A(other.A(x))  # (A' = A_1 A_2)
-        noise = self.noise_model
-        sensor = self.sensor_model
-        return Physics(
-            A=A,
-            noise_model=noise,
-            sensor_model=sensor,
+        return ComposedPhysics(
```

Comment:
I think we need to add the `compose` function below to handle combinations of linear + linear = linear, non_linear + linear = non_linear, and non_linear + non_linear = non_linear
```suggestion
        return compose([other, self])
```
with 
```
def compose(*physics: Union[Physics, LinearPhysics]):
    r"""
    Composes multiple forward operators :math:`A = A_1\circ A_2\circ \dots \circ A_n`.

    The measurements produced by the resulting model are :class:`deepinv.utils.TensorList` objects, where
    each entry corresponds to the measurements of the corresponding operator.

    :param deepinv.physics.Physics physics: Physics operators :math:`A_i` to be composed.
    """
    if all(isinstance(phys, LinearPhysics) for phys in physics):
        return ComposedLinearPhysics(physics)
    else:
        return ComposedPhysics(physics)
```

Reply:
Good point. Many thanks Julian!